packages/sanity/src/core/releases/tool/detail/documentTable/DocumentTableColumnDefs.tsx

```diff
@@ -57,21 +58,39 @@ export const getDocumentTableColumnDefs: (
         <Headers.BasicHeader text={t('table-header.action')} />
       </Flex>
     ),
-    cell: ({cellProps, datum}) => (
-      <Flex align="center" {...cellProps}>
-        <Box paddingX={2}>
-          {datum.document.publishedDocumentExists ? (
+    cell: ({cellProps, datum}) => {
+      const willBeUnpublished = isGoingToUnpublish(datum.document)
+      const BadgeToRender = () => {
```

Comment:
NIT: hmm, this approach of using the JSX component syntax to render `<BadgeToRender />` inside a component already irks me a little. Wonder if we could just have this function be called `actionBadge` and then 
```
        <Flex align="center" {...cellProps}>
          <Box paddingX={2}>
            {actionBadge}
          </Box>
        </Flex>
```
Non-blocking though

Reply:
Fair enough, I'll make the change!